### ISSUE TITLE
Add warnings when using request / response config with LAMBDA-PROXY in API Gateway

### ISSUE THREAD

**Issue by pmuens** [serverless/serverless#2220]
# This is a Feature Proposal

## Description

#2185 (corresponding issue: #2174) implements the new API Gateway lambda proxy functionality. The proxy simple forwards the request directly to the corresponding Lambda. The request and response configuration one can setup won't be considered anymore.

We should print a warning on the console that those configurations are ignored if a user uses the new lambda proxy in combination with the request / response syntax.

/cc @flomotlik @eahefnawy

**Comment by flomotlik** [comment]
agreed, it should be a warning and not an exception!

**Comment by dougmoscrop** [comment]
I believe this is fixed just FYI

**Comment by pmuens** [comment]
Done with #2185 /cc @flomotlik

**Comment by mhrisse** [comment]
Hi! I just ran into this problem, which makes my lambda functions not work anymore. 

```
{
message: "Internal server error"
}
```

However, I think I used this on rc1/rc2 before and it worked fine before.

```
  render:
    handler: react.render
    responses:
      default:
        statusCode: '200'
    events:
      - http:
          method: get
          path: app/v1/render
          response:
            headers:
              Content-Type: "'text/html'"
            template: $input.path('$')
```

From a developer ergonomics point of view printing a warning is great, however, maybe the warning could also point to the documentation that explains what the problem is? (or if you have a pointer for me here?) 

React and other libraries do that and this would help developer a lot. (I think struggling with serverless.yml config is one of the key concept to get used to when starting out with serverless). 

Thank you for all the good work and congrats to 1.0 and the funding!

*Matt
